### PR TITLE
Use an adjusted OpenLayers eslint config

### PR DESCRIPTION
**Pull request by marcjansen**
This PR suggests to lint our source code with an adjusted version of the openlayers eslint configuration.

GeoExt and OpenLayers itself use that particular configuration.

This makes our code easier to comprehend, and also revealed some issues, most of which were tagged with `TODO`.comments.

Most changes should be cosmetic, but the color util got a bigger makeover, but also some tests.

Please review.

PS: I added this in that many commits because…

* this makes it slightly easier to review
* this makes it easier to revert / adjust in case I introduced a bug or sth,
* I get paid by commit :stuck_out_tongue_winking_eye:

### CODE REVIEW

File: src/view/button/Measure.js [hunk 1]
```diff
@@ -408,16 +408,18 @@ Ext.define('BasiGX.view.button.Measure', {
...
+            btnText = '{textline}';
+            break;
+        case 'polygon':
+            btnText = '{textpoly}';
+            break;
+        case 'angle':
+            btnText = '{textangle}';
+            break;
+        default:
+            break;
```
Comment (dnlkoch) [inline src/view/button/Measure.js]: Add/Replace `"indent": ["error", 2, { "SwitchCase": 1 }]` in `.eslintrc` to ensure "correct" indent.

File: src/view/button/Measure.js [hunk 2]
```diff
@@ -460,14 +462,17 @@ Ext.define('BasiGX.view.button.Measure', {
 
     /**
      * Adds a tooltip on click where a measuring stop occured.
+     *
+     * @param {ol.MapBrowserEvent} evt The event which contains teh coordinate
```
Comment (dnlkoch) [inline src/view/button/Measure.js]: Typo: `teh` --> `the`.

File: src/view/button/Measure.js [hunk 3]
```diff
@@ -547,7 +557,7 @@ Ext.define('BasiGX.view.button.Measure', {
     /**
      * Handle pointer move by updating and repositioning the dynamic tooltip.
      *
-     * @param {ol.MapBrowserEvent} evt
+     * @param {ol.MapBrowserEvent} evt The Event from the pointermove.
```
Comment (dnlkoch) [inline src/view/button/Measure.js]: Typo: `Event` --> `event`.

File: src/view/container/NominatimSearch.js [hunk 4]
```diff
@@ -153,9 +165,16 @@ Ext.define("BasiGX.view.container.NominatimSearch", {
             })
         }),
 
-       /**
-        *
-        */
+        /**
+         * Returns an array of styles.
+         *
+         * TODO Are we accessing aprvate property here? `getSource().distance_`
```
Comment (dnlkoch) [inline src/view/container/NominatimSearch.js]: Typo: `aprvate` --> `private`.

File: src/view/container/NominatimSearch.js [hunk 5]
```diff
@@ -512,43 +543,56 @@ Ext.define("BasiGX.view.container.NominatimSearch", {
...
-        if(!Ext.Array.contains(featureExtent, Infinity)){
+        if (!Ext.Array.contains(featureExtent, Infinity)) {
             me.zoomToExtent(featureExtent);
         }
     },
 
     /**
      * Works with extent or geom.
+     *
+     * @param {ol.Extent|ol.geom.SimpleGeometry} extent The extent or geoemetry
```
Comment (dnlkoch) [inline src/view/container/NominatimSearch.js]: Typo: `geoemetry` --> `geometry`.

File: src/view/container/OverpassSearch.js [hunk 6]
```diff
@@ -164,9 +176,16 @@ Ext.define("BasiGX.view.container.OverpassSearch", {
             })
         }),
 
-       /**
-        *
-        */
+        /**
+         * Returns an array of styles.
+         *
+         * TODO Are we accessing aprvate property here? `getSource().distance_`
```
Comment (dnlkoch) [inline src/view/container/OverpassSearch.js]: Typo: `aprvate` --> `private`.

File: src/view/container/OverpassSearch.js [hunk 7]
```diff
@@ -557,97 +584,116 @@ Ext.define("BasiGX.view.container.OverpassSearch", {
...
-        if(!Ext.Array.contains(featureExtent, Infinity)){
+        if (!Ext.Array.contains(featureExtent, Infinity)) {
             me.zoomToExtent(featureExtent);
         }
     },
 
     /**
      * Works with extent or geom.
+     *
+     * @param {ol.Extent|ol.geom.SimpleGeometry} extent The extent or geoemetry
```
Comment (dnlkoch) [inline src/view/container/OverpassSearch.js]: Typo: `geoemetry` --> `geometry`

File: src/view/container/RedlineStyler.js [hunk 8]
```diff
@@ -786,24 +843,28 @@ Ext.define("BasiGX.view.container.RedlineStyler", {
     },
 
     /**
+     * Gets an array of icon sytle attributes from the form elements to
```
Comment (dnlkoch) [inline src/view/container/RedlineStyler.js]: Typo: `sytle` --> `style`

File: src/view/container/Redlining.js [hunk 9]
```diff
@@ -138,648 +138,685 @@ Ext.define("BasiGX.view.container.Redlining", {
...
-           fn: function(decision, text) {
-               if (decision === "ok") {
-                   if (text.length > me.postitTextMaxLength) {
-                       BasiGX.confirm(me.getViewModel().get(
+        var button = Ext.ComponentQuery.query('button[name=postitbutton]')[0];
+        button.toggle(false);
+    },
+
+    /**
+     * Modify a postits text.
```
Comment (dnlkoch) [inline src/view/container/Redlining.js]: Typo: `postits` --> `postit`.

File: src/view/container/WfsSearch.js [hunk 10]
```diff
@@ -156,9 +167,16 @@ Ext.define("BasiGX.view.container.WfsSearch", {
             })
         }),
 
-       /**
-        *
-        */
+        /**
+         * Returns an array of styles.
+         *
+         * TODO Are we accessing aprvate property here? `getSource().distance_`
```
Comment (dnlkoch) [inline src/view/container/WfsSearch.js]: Typo: `aprvate` --> `private`.

File: src/view/container/WfsSearch.js [hunk 11]
```diff
@@ -409,12 +433,13 @@ Ext.define("BasiGX.view.container.WfsSearch", {
     },
 
     /**
-     *
+     * Issues a `DescribeFeatureType` request, so that we know details about the
+     * Featuretyoe we are working on.
```
Comment (dnlkoch) [inline src/view/container/WfsSearch.js]: Typo: `Featuretyoe` --> `featuretype`.

File: src/view/form/Print.js [hunk 12]
```diff
@@ -748,44 +808,51 @@ Ext.define("BasiGX.view.form.Print", {
...
+            break;
+        case 'String':
+            attributeFields = me.getStringField(attributeRec);
+            break;
+        case 'DataSourceAttributeValue':
+            Ext.toast('Data Source not yet supported');
+            attributeFields = me.getStringField(attributeRec);
+            break;
+        default:
+            break;
```
Comment (dnlkoch) [inline src/view/form/Print.js]: See above (somewhere :-) ).

File: src/view/form/field/GazetteerCombo.js [hunk 13]
```diff
@@ -54,22 +54,23 @@ Ext.define('BasiGX.view.form.field.GazetteerCombo',{
     },
 
     /**
-     * @param combo
-     * @param newValue
+     * When the combo value changes, do a ne w Gazetteer search.
```
Comment (dnlkoch) [inline src/view/form/field/GazetteerCombo.js]: Typo: `ne w` --> `new `.
Typo: `Gazetteer` --> `gazetteer`.

File: src/view/form/field/MultiSearchCombo.js [hunk 14]
```diff
@@ -150,7 +150,10 @@ Ext.define('BasiGX.view.form.field.MultiSearchCombo',{
 
     },
 
-    onBoxReady: function(){
+    /**
+     * Ensures that the value is clearer when one hits escape.
```
Comment (dnlkoch) [inline src/view/form/field/MultiSearchCombo.js]: Typo: `clearer` --> `cleared`.

File: src/view/panel/GraphicPool.js [hunk 15]
```diff
@@ -249,23 +251,31 @@ Ext.define("BasiGX.view.panel.GraphicPool", {
...
+            failure: function() {
+                BasiGX.error(
+                    me.getViewModel().get('graphicDeleteInfoError')
+                );
             }
         });
     },
 
     /**
+     * Returns the Ext.data.Model instacne for teh selected picture.
```
Comment (dnlkoch) [inline src/view/panel/GraphicPool.js]: Typo: `instacne` --> `instance`.
Typo: `teh` --> `the`.

File: src/view/panel/LegendTree.js [hunk 16]
```diff
@@ -218,38 +218,46 @@ Ext.define("BasiGX.view.panel.LegendTree", {
     },
 
     /**
+     * Recursively seacrhes the passed record (or its children) for a
```
Comment (dnlkoch) [inline src/view/panel/LegendTree.js]: Typo: `seacrhes` --> `searches`.

File: src/view/panel/LegendTree.js [hunk 17]
```diff
@@ -218,38 +218,46 @@ Ext.define("BasiGX.view.panel.LegendTree", {
...
         return color;
     },
 
     /**
      * Method gives access to the rows style.
+     *
      * If a layer is configured with property 'treeColor', the color will
      * get applied here. Folders will inherit the color
+     *
+     * @param {Ext.data.Model} rec The tree node recordof the row.
```
Comment (dnlkoch) [inline src/view/panel/LegendTree.js]: Typo: `recordof` --> `record of`.

**Comment by dnlkoch** [comment]
Thanks @marcjansen for this outstanding work!

File: src/util/Color.js [hunk 18]
```diff
@@ -21,90 +21,140 @@ Ext.define('BasiGX.util.Color', {
...
-         * Method converts a hex8 color string into an rgba color string.
+         * This method converts a hex color string into an rgba color string.
          *
          * Example:
+         *
          *     var rgba = BasiGX.util.Color.hexToRgba("#ff0000", 0.5);
          *     // rgba is now: "rgba(255,0,0,0.5)"
+         *
+         * @param {String} hex The hex string to convert.
+         * @param {Number} opacity The opacity to set, should be between `` and
```
Comment (marcjansen) [inline src/util/Color.js]: Done.

File: src/util/ConfigParser.js [hunk 19]
```diff
@@ -484,21 +512,24 @@ Ext.define('BasiGX.util.ConfigParser', {
...
-                // as descibed on https://wiki.intranet.terrestris.de/doku.php?id=clustering
-                // There is currently now way in ol3 to request features on every
-                // extent change, so we need to handle it ourselves with map listeners,
-                // which happens in the cluster plugin
-                if (config.type === "WFSCluster") {
+                // The wfscluster type expects a geoserver view similar
+                // as described on https://wiki.intranet.terrestris.de/doku.php
+                // ?id=clustering
+                //
+                // There is currently now way in ol3 to request features on
```
Comment (marcjansen) [inline src/util/ConfigParser.js]: Done.

File: src/util/Map.js [hunk 20]
```diff
@@ -73,36 +74,54 @@ Ext.define('BasiGX.util.Map', {
             return (resolution * mpu * inchesPerMeter * dpi);
         },
 
-        getResolution: function (map) {
+        /**
+         * Returns the resolution of the passed maps view.
```
Comment (marcjansen) [inline src/util/Map.js]: Done.

File: src/util/Map.js [hunk 21]
```diff
@@ -73,36 +74,54 @@ Ext.define('BasiGX.util.Map', {
...
+         *
+         * @param {ol.Map} map The map to get the resolution of.
```
Comment (marcjansen) [inline src/util/Map.js]: Done.

File: src/util/Url.js [hunk 22]
```diff
@@ -44,33 +47,44 @@ Ext.define('BasiGX.util.Url', {
...
         },
 
         /**
-         * Return the name of the web project like this
+         * Return the base URL of the web project.
          *
-         *   Tribulus
+         * TODO Are we sure we have an application always deployed like it is
+         *      assumed herein?
+         *     'folder').
```
Comment (marcjansen) [inline src/util/Url.js]: Done.

File: src/ux/ContextSensitiveHelp.js [hunk 23]
```diff
@@ -130,24 +148,29 @@ Ext.define('BasiGX.ux.ContextSensitiveHelp', {
         },
 
         /**
+         * Determines the component at the passed pointb that has help info
```
Comment (marcjansen) [inline src/ux/ContextSensitiveHelp.js]: Done.

File: src/view/button/Measure.js (hunk 1 above)
Comment (marcjansen) [inline src/view/button/Measure.js]: Done.

File: src/view/button/Measure.js (hunk 2 above)
Comment (marcjansen) [inline src/view/button/Measure.js]: Done.

File: src/view/button/Measure.js (hunk 3 above)
Comment (marcjansen) [inline src/view/button/Measure.js]: Done.

File: src/view/container/NominatimSearch.js (hunk 4 above)
Comment (marcjansen) [inline src/view/container/NominatimSearch.js]: Done.

File: src/view/container/NominatimSearch.js (hunk 5 above)
Comment (marcjansen) [inline src/view/container/NominatimSearch.js]: Done.

File: src/view/container/OverpassSearch.js (hunk 6 above)
Comment (marcjansen) [inline src/view/container/OverpassSearch.js]: Done.

File: src/view/container/OverpassSearch.js (hunk 7 above)
Comment (marcjansen) [inline src/view/container/OverpassSearch.js]: Done.

File: src/view/container/RedlineStyler.js (hunk 8 above)
Comment (marcjansen) [inline src/view/container/RedlineStyler.js]: Done.

File: src/view/container/Redlining.js (hunk 9 above)
Comment (marcjansen) [inline src/view/container/Redlining.js]: Done.

File: src/view/container/WfsSearch.js (hunk 10 above)
Comment (marcjansen) [inline src/view/container/WfsSearch.js]: Done.

File: src/view/container/WfsSearch.js (hunk 11 above)
Comment (marcjansen) [inline src/view/container/WfsSearch.js]: Done.

File: src/view/form/Print.js (hunk 12 above)
Comment (marcjansen) [inline src/view/form/Print.js]: Done.

File: src/view/form/field/GazetteerCombo.js (hunk 13 above)
Comment (marcjansen) [inline src/view/form/field/GazetteerCombo.js]: Done.

File: src/view/form/field/MultiSearchCombo.js (hunk 14 above)
Comment (marcjansen) [inline src/view/form/field/MultiSearchCombo.js]: Done.

File: src/view/panel/GraphicPool.js (hunk 15 above)
Comment (marcjansen) [inline src/view/panel/GraphicPool.js]: Done.

File: src/view/panel/LegendTree.js (hunk 16 above)
Comment (marcjansen) [inline src/view/panel/LegendTree.js]: Done.

File: src/view/panel/LegendTree.js (hunk 17 above)
Comment (marcjansen) [inline src/view/panel/LegendTree.js]: Done.

**Comment by marcjansen** [comment]
Thanks for the tedious review work, @dnlkoch. I have addressed all your comments and will merge once everything is green.

Thanks again for the review!

**Pull request merged**